Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1

```diff
@@ -1,111 +1,117 @@
+    $testMachines               = Get-VstsInput -Name testMachines
+    $runAsProcess               = Get-VstsInput -Name runAsProcess
+    $machineUserName            = Get-VstsInput -Name machineUserName -Require
+    $machinePassword            = Get-VstsInput -Name machinePassword -Require
+    $agentLocation              = Get-VstsInput -Name agentLocation
+    $updateTestAgent            = Get-VstsInput -Name updateTestAgent
+    $isDataCollectionOnly       = Get-VstsInput -Name isDataCollectionOnly
+    $testPlatform               = Get-VstsInput -Name testPlatform
+
+    # If Run as process (Run UI Tests) is true both autologon and disable screen saver needs to be true.
+    $logonAutomatically = $runAsProcess
+    $disableScreenSaver = $runAsProcess
+
+    Write-Host "****************************************************************"
+    Write-Host "                    Task Input Information                      "
+    Write-Host "----------------------------------------------------------------"
+    Write-Host "testMachineInput         = $testMachineGroup"
+    Write-Host "adminUserName            = $adminUserName"
+    Write-Host "machineUserName          = $machineUserName"
+    Write-Host "WinRmProtocal            = $winRmProtocol"
+    Write-Host "testCertificate          = $testCertificate"
+    Write-Host "resourceFilteringMethod  = $resourceFilteringMethod"
+    Write-Host "filter testMachines      = $testMachines"
+    Write-Host "runAsProcess             = $runAsProcess"
+    Write-Host "logonAutomatically       = $logonAutomatically"
+    Write-Host "disableScreenSaver       = $disableScreenSaver"
+    Write-Host "updateTestAgent          = $updateTestAgent"
+    Write-Host "isDataCollectionOnly     = $isDataCollectionOnly"
+    Write-Host "testPlatform             = $testPlatform"
+    Write-Host "agentLocation            = $agentLocation"
+    Write-Host "****************************************************************"
+
+    $downloadTestAgentScript            = "$PSScriptRoot\DownloadTestAgent.ps1"
+    $setupTestMachineForUITestsScript   = "$PSScriptRoot\SetupTestMachineForUITests.ps1"
+    $TestAgentConfigurationScript       = "$PSScriptRoot\TestAgentConfiguration.ps1"
+    $testAgentHelperScript              = "$PSScriptRoot\TestAgentHelper.ps1"
+    $installTestAgentScript             = "$PSScriptRoot\InstallTestAgent.ps1"
+    $verifyTestAgentInstalledScript     = "$PSScriptRoot\VerifyTestAgentInstalled.ps1"
+
+    # Fix Assembly Redirections
+    # VSTS uses Newton Json 8.0 while the System.Net.Http uses 6.0
+    # Redirection to Newton Json 8.0
+    $jsonAssembly = [reflection.assembly]::LoadFrom($PSScriptRoot + "\modules\Newtonsoft.Json.dll") 
+    $onAssemblyResolve = [System.ResolveEventHandler] {
+        param($sender, $e)
+        if ($e.Name -eq "Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed") { return $jsonAssembly }
```

Comment:
>$e.Name -eq "Newtonsoft.Json, Version=6.0.0.0 [](start = 12, length = 45)

any reason to have it only for 6.0//? can we load the 8.0 version for everything..